mempool/src/shared_mempool.rs

```diff
@@ -173,14 +199,42 @@ async fn sync_with_peers<'a>(
     }
 }
 
-/// used to validate incoming transactions and add them to local Mempool
+fn convert_txn_from_proto(txn_proto: SignedTransactionProto) -> Option<SignedTransaction> {
+    match SignedTransaction::try_from(txn_proto.clone()) {
+        Ok(txn) => Some(txn),
+        Err(e) => {
+            security_log(SecurityEvent::InvalidTransactionMP)
+                .error(&e)
+                .data(&txn_proto)
+                .log();
+            None
+        }
+    }
+}
+
+/// submits a list of SignedTransaction to the local mempool,
+/// and returns an AdmissionControlStatus if `is_ac_endpoint_submission`
+/// if `is_ac_endpoint_submission`, we expect a single transaction to be submitted
 async fn process_incoming_transactions<V>(
     smp: SharedMempool<V>,
-    peer_id: PeerId,
+    peer_id: Option<PeerId>,
     transactions: Vec<SignedTransaction>,
-) where
+) -> Option<Status>
```

Comment:
this interface looks a bit weird with optional PeerId and status and a lot of branches. Can we somehow clean it up? Maybe return vector by default or pass callback here?

Reply:
yeah... this is because I'm trying not to duplicate code for adding transactions to mempool for client submissions and in-network submissions, and with client submissions, I need the result. The tradeoff here is code duplication vs. cleaner interface - if cleaner interface is more important I can write separate methods for both use cases. In the current state I can return status as vector by default, but passing in callback wouldn't be applicable for in-network submission, so that would have to be an optional parameter